src/Catalyst.jl

```diff
@@ -68,6 +70,12 @@ export @reaction_network, @add_reactions, @reaction, @species
 include("registered_functions.jl")
 export mm, mmr, hill, hillr, hillar
 
+# spatial reaction networks
+include("lattice_reaction_system_diffusion.jl")
+export DiffusionReaction, diffusion_reactions, isdiffusionparameter
```

Comment:
The idea is that these only do diffusion. Currently I define
```
struct DiffusionParameter end
...
struct DiffusionReaction <: AbstractSpatialReaction
   ...
end
```
and the idea was to later on, if we want, create other `AbstractSpatialReaction`s for other types of such reactions.
(happy to change approach though)

Reply:
But `A_i --> A_j` at some rate encompasses more than just approximations to diffusion. It is also how one can approximate advection and drift-diffusion. So what you have is already more general than the name.

Reply:
Ok, might be me just not familiar with terminology. I thought A moving to an adjacent compartment at linear rate was diffusion. Would renaming `DiffusionReaction` to `TransportReaction` and still make that a subtype of `AbstractSpatialReaction` make sense?

Reply:
Sure, that sounds good. Technically `A_i -> A_j` isn't diffusion at all since it is discrete motion on a graph/lattice and diffusion is continuous motion in continuous space. It can, however, for appropriate choices of the rate represent a random walk, which can be a convergent approximation to diffusion as one shrinks the edge lengths to zero.

Reply:
All references to "diffusion" have now been removed. Currently, we use transport reactions (where a species is moved at contact rate between adjacent compartments), and spatial more generally for this kind of stuff.